`west build` compilation failed

### Describe the bug

The commander line `west build -p always -b mimxrt1170_evk@B/mimxrt1176/cm7 tests/bluetooth/shell --extra-conf=prj_br.conf  -d shell_br` is failed to execute.
The zephyr version is 7fd932b9357e198dfae9c2eaf45df73e7cfd4b1a.

I found the issue is caused by the change https://github.com/zephyrproject-rtos/zephyr/pull/108282.
If I remove the `f'-DWEST_TOPDIR={west_topdir(self.source_dir)}'` from `final_cmake_args`, the commander can pass to build.


### Regression

- [x] This is a regression.

### Steps to reproduce

Steps to reproduce the behavior:
1. west build -p always -b mimxrt1170_evk@B/mimxrt1176/cm7 tests/bluetooth/shell --extra-conf=prj_br.conf  -d shell_br
2. See error

### Relevant log output

```shell
-- west build: making build dir C:\_DDM\zephyrproject\zephyr\shell_br pristine
-- west build: generating a build system
Loading Zephyr default modules (Zephyr base).
-- Application: C:/_DDM/zephyrproject/zephyr/tests/bluetooth/shell
-- CMake version: 3.30.5
-- Found Python3: C:/Python314/python.exe (found suitable version "3.14.2", minimum required is "3.12") found components: Interpreter
-- Cache files will be written to: C:/_DDM/zephyrproject/zephyr/.cache
-- Zephyr version: 4.4.99 (C:/_DDM/zephyrproject/zephyr)
-- Found west (found suitable version "1.5.0", minimum required is "0.14.0")
-- Board: mimxrt1170_evk, Revision: B, qualifiers: mimxrt1176/cm7
-- ZEPHYR_TOOLCHAIN_VARIANT not set, trying to locate Zephyr SDK
-- Found host-tools: zephyr 1.0.0 (C:/_DDM/zephyr-sdk-1.0.0)
-- Found toolchain: zephyr 1.0.0 (C:/_DDM/zephyr-sdk-1.0.0)
-- Found Dtc: C:/ProgramData/chocolatey/bin/dtc.exe (found suitable version "1.5.0", minimum required is "1.4.6")
-- Found BOARD.dts: C:/_DDM/zephyrproject/zephyr/boards/nxp/mimxrt1170_evk/mimxrt1170_evk_mimxrt1176_cm7.dts
-- Found devicetree overlay: C:/_DDM/zephyrproject/zephyr/boards/nxp/mimxrt1170_evk/mimxrt1170_evk_mimxrt1176_cm7_B.overlay
-- Found devicetree overlay: C:/_DDM/zephyrproject/zephyr/tests/bluetooth/shell/boards/mimxrt1170_evk_mimxrt1176_cm7_B.overlay
-- Generated zephyr.dts: C:/_DDM/zephyrproject/zephyr/shell_br/zephyr/zephyr.dts
-- Generated pickled edt: C:/_DDM/zephyrproject/zephyr/shell_br/zephyr/edt.pickle
-- Generated devicetree_generated.h: C:/_DDM/zephyrproject/zephyr/shell_br/zephyr/include/generated/zephyr/devicetree_generated.h

warning: MINIMAL_LIBC_STRING_ERROR_TABLE (defined at lib/libc\minimal/Kconfig:53) was assigned the
value 'y' but got the value 'n'. Check these unsatisfied dependencies: MINIMAL_LIBC (=n). See
http://docs.zephyrproject.org/latest/kconfig.html#CONFIG_MINIMAL_LIBC_STRING_ERROR_TABLE and/or look
up MINIMAL_LIBC_STRING_ERROR_TABLE in the menuconfig/guiconfig interface. The Application
Development Primer, Setting Configuration Values, and Kconfig - Tips and Best Practices sections of
the manual might be helpful too.


warning: Not secure symbol PSA_WANT_ALG_MD5 is enabled.

Parsing C:/_DDM/zephyrproject/zephyr/Kconfig
Loaded configuration 'C:/_DDM/zephyrproject/zephyr/boards/nxp/mimxrt1170_evk/mimxrt1170_evk_mimxrt1176_cm7_defconfig'
Merged configuration 'C:/_DDM/zephyrproject/zephyr/tests/bluetooth/shell/prj.conf'
Merged configuration 'C:/_DDM/zephyrproject/zephyr/tests/bluetooth/shell/boards/mimxrt1170_evk_mimxrt1176_cm7_B.conf'
Merged configuration 'C:/_DDM/zephyrproject/zephyr/tests/bluetooth/shell/prj_br.conf'
Configuration saved to 'C:/_DDM/zephyrproject/zephyr/shell_br/zephyr/.config'
Kconfig header saved to 'C:/_DDM/zephyrproject/zephyr/shell_br/zephyr/include/generated/zephyr/autoconf.h'
-- Found GnuLd: C:/_DDM/zephyr-sdk-1.0.0/gnu/arm-zephyr-eabi/arm-zephyr-eabi/bin/ld.bfd.exe (found version "2.43.1")
-- The C compiler identification is GNU 14.3.0
-- The CXX compiler identification is GNU 14.3.0
-- The ASM compiler identification is GNU
-- Found assembler: C:/_DDM/zephyr-sdk-1.0.0/gnu/arm-zephyr-eabi/bin/arm-zephyr-eabi-gcc.exe
CMake Error at C:/Program Files/CMake/share/cmake-3.30/Modules/CheckCCompilerFlag.cmake:51 (cmake_check_compiler_flag):
  Syntax error in cmake code at

    C:/Program Files/CMake/share/cmake-3.30/Modules/CheckCCompilerFlag.cmake:51

  when parsing string

    -fmacro-prefix-map=C:\_DDM\zephyrproject=WEST_TOPDIR

  Invalid character escape '\z'.
Call Stack (most recent call first):
  C:/_DDM/zephyrproject/zephyr/cmake/modules/extensions.cmake:2467 (check_c_compiler_flag)
  C:/_DDM/zephyrproject/zephyr/cmake/modules/extensions.cmake:1245 (check_compiler_flag)
  C:/_DDM/zephyrproject/zephyr/cmake/modules/extensions.cmake:2506 (zephyr_check_compiler_flag)
  C:/_DDM/zephyrproject/zephyr/cmake/modules/extensions.cmake:2482 (target_cc_option_fallback)
  C:/_DDM/zephyrproject/zephyr/cmake/modules/extensions.cmake:155 (target_cc_option)
  C:/_DDM/zephyrproject/zephyr/CMakeLists.txt:637 (zephyr_cc_option)


-- Configuring incomplete, errors occurred!
FATAL ERROR: command exited with status 1: 'C:\Program Files\CMake\bin\cmake.EXE' -DWEST_PYTHON=C:/Python314/python.exe '-DWEST_TOPDIR=C:\_DDM\zephyrproject' -DWEST_VERSION=1.5.0 '-BC:\_DDM\zephyrproject\zephyr\shell_br' -GNinja '-DWEST_PYTHON_PROPERTIES="Python;3;14;2;64;;cp314-win_amd64.pyd;;C:\Python314\Lib;C:\Python314\Lib;C:\Python314\Lib\site-packages;C:\Python314\Lib\site-packages"' -DBOARD=mimxrt1170_evk@B/mimxrt1176/cm7 -DEXTRA_CONF_FILE=prj_br.conf -DCMAKE_EXPORT_COMPILE_COMMANDS=ON '-SC:\_DDM\zephyrproject\zephyr\tests\bluetooth\shell'
```

### Impact

Major – Severely degrades functionality; workaround is difficult or unavailable.

### Environment

- OS: (Windows 11)
- Toolchain: (Zephyr SDK)
- Commit SHA: 7fd932b9357e198dfae9c2eaf45df73e7cfd4b1a


### Additional Context

_No response_

--- a/scripts/west_commands/build.py
+++ b/scripts/west_commands/build.py
@@ -702,7 +702,7 @@ class Build(Forceable):
         #
         # west build -- -DOVERLAY_CONFIG=relative-path.conf
         final_cmake_args = [f'-DWEST_PYTHON={pathlib.Path(sys.executable).as_posix()}',
-                            f'-DWEST_TOPDIR={west_topdir(self.source_dir)}',
+                            f'-DWEST_TOPDIR={pathlib.Path(str(west_topdir(self.source_dir))).as_posix()}',
                             f'-DWEST_VERSION={str(__version__)}',
                             f'-B{self.build_dir}',
                             f'-G{config_get("generator", DEFAULT_CMAKE_GENERATOR)}']

